### PR TITLE
LANG-1258: Add ArrayUtils#toStringArray(Object[]) method

### PR DESCRIPTION
**Pull request by PascalSchumacher**
patch supplied by IG

### CODE REVIEW

**Comment by kinow** [comment]
Patch looks good. I wonder if the inline if will raise a warning in checkstyle. Other than that, +1 :D

**Comment by Xaerxess** [comment]
[I added null elements handling in PR to this branch](https://github.com/PascalSchumacher/commons-lang/pull/1), because in current state NPE would be thrown in case of null in array.

**Comment by PascalSchumacher** [comment]
I have updated the pull request with @Xaerxess changes.

File: src/main/java/org/apache/commons/lang3/ArrayUtils.java [hunk 1]
```diff
@@ -7967,4 +7967,29 @@ public static boolean isSorted(boolean[] array) {
 
         return removeAll(array, Arrays.copyOf(indices, count));
     }
+
+    /**
+     * Returns an array containing the string representation of each entry in the argument array.
+     *
+     * <p>
+     * A {@code null} input Object[] returns {@code null}.
```
Comment (Xaerxess) [inline src/main/java/org/apache/commons/lang3/ArrayUtils.java]: One more thing came to my mind - we can document `null` elements handling. I don't have exact words for that, but something like `A {@code null} input Object[] returns {@code null}. {@code null} elements are preserved.`.

**Comment by Xaerxess** [comment]
Sure, consistency in API is the key. But `stringIfNull` is used when array itself is `null`, not element, so it wouldn't be consistent with the API.

**Comment by PascalSchumacher** [comment]
@Xaerxess Sorry for the confusion, I wasn't talking about `toString`, but about the `toPrimitive` methods e.g.

https://github.com/apache/commons-lang/blob/96c8ea2fb3719e2f6e3d7a4d7b46718f26515a86/src/main/java/org/apache/commons/lang3/ArrayUtils.java#L4497

As these also convert the type of the array I think they are the most similar existing methods (compared to the new method in this pull request).

**Comment by PascalSchumacher** [comment]
Merged: https://github.com/apache/commons-lang/commit/8d95ae41975a2307501aa0f4a7eba296c59edce9 and https://github.com/apache/commons-lang/commit/8d601ab71228f7c3dff950540e7ee6e4043e9053

Thanks everybody!

**Pull request closed without merging**